### PR TITLE
Delete Old Logs

### PR DESCRIPTION
**Pull request by ohmcodes**
Config - MaxAgeInSeconds 86400 default 24hrs

### CODE REVIEW

File: AsaApi/main.cpp [hunk 1]
```diff
@@ -63,6 +63,43 @@ void OpenConsole()
 	SetConsoleOutputCP(CP_UTF8);
 }
 
+void DeleteOldLogFiles()
+{
+	const std::string config_path = AsaApi::Tools::GetCurrentDir() + "/config.json";
```
Comment (Lethalinjectionx) [inline AsaApi/main.cpp:68]: Using the config example I mentioned in config.json. This format allows for the config file to be missing and not break existing installs of the API while maintaining the current functionality of not pruning logs. Accessing the config has you have it currently will crash if they do not update the config.

`if (config.value("DeleteOldLogs", nlohmann::json::object()).value("Enabled", false) == false)
   return;
   
   const int maxAgeInSeconds = config.value("DeleteOldLogs", nlohmann::json::object()).value("MaxAgeInHours", 24) * 3600
`

**Comment by Pelayori** [comment]
Not sure if broadcasting the deletion of a log is relevant to the average user.

**Review by Lethalinjectionx — CHANGES_REQUESTED** [review]

File: Configs/config.json [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-    "AutomaticPluginReloading":true,
-    "AutomaticPluginReloadSeconds":5,
+{
+  "settings": {
+    "AutomaticPluginReloading": true,
+    "AutomaticPluginReloadSeconds": 5,
     "SaveWorldBeforePluginReload": true,
-    "AttachToParent": true
+    "AttachToParent": true,
+    "MaxAgeInSeconds": 86400
```
[the remark this answers is not in the document]
Comment (Pelayori) [reply Configs/config.json:7]: This config format looks clearer for me, too.

[the remark this answers is not in the document]
Comment (SubstituteR) [reply Configs/config.json:7]: I'm not sold on hours for max age, but it's fine.

Personally I'm a fan of either
```
"DeleteOldLogs": { "Enabled": true, "MaxAge": 86400 }
```
or
```
"DeleteOldLogs": { "Enabled": true, "MaxAge": "1D"}
```

[the remark this answers is not in the document]
Comment (Pelayori) [reply Configs/config.json:7]: `"DeleteOldLogs": { "Enabled": true, "MaxAge": "24H"}` is also good for me.

[the remark this answers is not in the document]
Comment (ohmcodes) [reply Configs/config.json:7]: maybe we stick to hours without H, i just pick seconds so it easily to set but not really user friendly as they need to multiply if they want hours, so for limiting the scope i will pick hours like 1 , 2.. up to 24 or more if they want days 48 for 2days and will add the Enable bool as well

**Pull request merged**